### ISSUE TITLE
V8 database create fails

### ISSUE THREAD

**Issue by sad-dev** [github/codeql-cli-binaries#110]
Compiling a debug build of V8 (tools/dev/gm.py x64.debug) results in an empty database and multiple errors of the form

```
Processed command line: ... extractor ...
Command-line error: language modes specified are incompatible

1 catastrophic error detected in this compilation.
```
The release build (tools/dev/gm.py x64.release) builds just fine.

**Comment by aeisenberg** [comment]
What version of codeql CLI are you running and what platform are you on? This looks like an issue with the C extractor. I'll notify the team who maintains this. Thank you for letting us know about this.

**Comment by sad-dev** [comment]
I have tried with both the 2.8.0 and 2.7.6 versions of the CLI; I was building on Ubuntu 20.04

**Comment by Manouchehri** [comment]
When did you notice this broke?

**Comment by aeisenberg** [comment]
Thanks. Our C extractor team is looking at this. If you have a log file to attach, that would be helpful.

**Comment by sad-dev** [comment]
It appears that when ``symbol_level = 2`` is set in args.gn, the build fails, regardless of whether it is on release or debug.
I am on 9226834 of V8 (24 Jan)

**Comment by Manouchehri** [comment]
Yep, I came to the same conclusion as well about the symbol level.

**Comment by sad-dev** [comment]
If it helps, the Compilation for ``obj/third_party/zlib/zlib/deflate.o``:

``symbol_level = 2:``
``-g2 -gdwarf-aranges -gsplit-dwarf -ggnu-pubnames -Xclang -fuse-ctor-homing ``

``symbol_level = 0:``
``-g0``

Running the full mimic command after removing ``-Xclang -fuse-ctor-homing`` no longer throws a catastrophic error (although it does report an error:
```
"...v8/v8/third_party/llvm-build/Release+Asserts/lib/clang/14.0.0/include/xmmintrin.h", line 3008: error: function call is not allowed in a constant expression
  #if defined(__SSE2__) && !__building_module(_Builtin_intrinsics)
                            ^
```

**Comment by sad-dev** [comment]
Hi, has this issue been resolved as of 2.8.4 ?

**Comment by jketema** [comment]
Hi @sad-dev, we have fixed this issue. The fix should be part of CodeQL 2.12.2, which should be released in the next few weeks.
